scripts/setup-env.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+CHECK="test \"$_\" != \"$0\""
+$CHECK || { echo >&2 "Use this script by sourcing it \`. $0\` instead"; exit 1; }
+
+find_keyfile() {
+  local address="$(tr '[:upper:]' '[:lower:]' <<<"${2#0x}")"
+  while IFS= read -r -d '' file; do
+    if [ "$(jq -r .address "$file")" == "$address" ]; then
+      echo "$file"
+      break
+    fi
+  done < <(find "$1" -type f -print0)
+}
+
+TESTNET_PORT=${TESTNET_PORT:-8545}
+TESTNET_HOST=${TESTNET_HOST:-localhost}
+TESTNET_URL="http://$TESTNET_HOST:$TESTNET_PORT"
+
+echo "Using RPC URL $TESTNET_URL. Change by setting TESTNET_PORT and TESTNET_HOST"
+
+# See if dapp testnet or parity dev chain is running
+
+if [ "$KEYSTORE_PATH" ]; then
+  true # If KEYSTORE_PATH is set don't look for running ethereum clients
+elif { pgrep -a geth && test -d "$HOME/.dapp/testnet/$TESTNET_PORT"; }; then
+  KEYSTORE_PATH="$HOME/.dapp/testnet/$TESTNET_PORT/keystore"
+  echo Found geth process
+elif { pgrep -a parity && test -d "$HOME/.local/share/io.parity.ethereum/keys/DevelopmentChain"; }; then
+  KEYSTORE_PATH="$HOME/.local/share/io.parity.ethereum/keys/DevelopmentChain"
+  echo Found parity process
+fi
```

Comment:
I mean an `else` here with

```bash
echo "No ethereum client found, set KEYSTORE_PATH and run script again" && exit 1
```

Reply:
Aah, of course. I think that's how I did it originally when it was a stand alone. The problem is that if you do `exit` when sourcing your shell will die :P

Reply:
True. There are [workarounds](https://unix.stackexchange.com/a/277350) but I think they're more trouble than they're worth.

Feel free to mark as resolved :)

Reply:
Yeah, could wrap it in a function and use `return`. This would feel a bit more contained also.

Reply:
You sure you need to wrap it in a function?

Reply:
No, you are right. `return` in top level terminates execution of sourced file without killing the shell.

Cool, will make the changes as you suggest!